python-checks/src/main/java/org/sonar/python/checks/cdk/PublicApiIsSecuritySensitiveCheck.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * SonarQube Python Plugin
+ * Copyright (C) 2011-2022 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.python.checks.cdk;
+
+import java.util.Optional;
+import java.util.function.Predicate;
+import org.sonar.check.Rule;
+import org.sonar.plugins.python.api.tree.Expression;
+import org.sonar.plugins.python.api.tree.StringLiteral;
+
+@Rule(key = "S6333")
+public class PublicApiIsSecuritySensitiveCheck extends AbstractCdkResourceCheck {
+  private static final String SAFE_API_MESSAGE = "Make sure that creating public APIs is safe here.";
+  private static final String OMITTING_MESSAGE = "Omitting \"authorization_type\" disables authentication. Make sure it is safe here.";
+
+  @Override
+  protected void registerFqnConsumer() {
+    checkFqn("aws_cdk.aws_apigateway.CfnMethod",
+      (subscriptionContext, callExpression) -> CdkUtils.getArgument(subscriptionContext, callExpression, "authorization_type").ifPresentOrElse(
+        argument -> argument.addIssueIf(isAuthorizationTypeNone(), SAFE_API_MESSAGE),
```

Comment:
```suggestion
        argument -> argument.addIssueIf(CdkPredicate.isString("NONE"), SAFE_API_MESSAGE),
```
There is already a predicate that performs the same action. Maybe you want to use this one.